velox/buffer/Buffer.h

```diff
@@ -381,34 +381,19 @@ class AlignedBuffer : public Buffer {
     auto oldCapacity = checkedPlus<size_t>(old->capacity(), kPaddedSize);
     auto preferredSize =
         pool->preferredSize(checkedPlus<size_t>(size, kPaddedSize));
-    // Make the buffer no longer owned by '*buffer' because reallocate
-    // may free the old buffer. Reassigning the new buffer to
-    // '*buffer' would be a double free.
+
+    void* newPtr = pool->reallocate(old, oldCapacity, preferredSize);
```

Comment:
`reallocate` is virtual and could return the same pointer for some implementation in the future.  Shall we not drop the optimization too early?

Reply:
We can have a test to enforce this behavior of reallocate?